lock-api/src/main/java/com/palantir/lock/client/TimeLockClient.java

```diff
@@ -96,7 +97,12 @@ public LockImmutableTimestampResponse lockImmutableTimestamp() {
     public StartIdentifiedAtlasDbTransactionResponse startIdentifiedAtlasDbTransaction() {
         StartIdentifiedAtlasDbTransactionResponse response = executeOnTimeLock(
                 delegate::startIdentifiedAtlasDbTransaction);
-        lockRefresher.registerLock(response.immutableTimestamp().getLock());
+        try {
+            lockRefresher.registerLock(response.immutableTimestamp().getLock());
+        } catch (Throwable t) {
+            unlock(ImmutableSet.of(response.immutableTimestamp().getLock()));
```

Comment:
`registerLock` (which in the future PR becomes `registerLocks`, i.e. a batched variant) essentially performs `addAll` on a `ConcurrentHashMap`. The only documented exceptions here are `NullPointerException` in the case when the lock is `null` (which will naturally throw when trying to `unlock` also), or `UnsupportedOperationException` which really should not throw unless something is really wrong.

So all in all, there is little-to-no chance that, in practice, this would happen anyway; however, to be technically correct we should do this (or if it takes on additional behaviour in the future). We probably don't want to retry because that probably means something really wrong has happened, and in that case we want that to propagate up.